RecoVertex/PrimaryVertexProducer/src/DAClusterizerInZT_vect.cc

```diff
@@ -0,0 +1,993 @@
+	 (niter++ < maxIterations_)) {}
+
+  // annealing loop, stop when T<Tmin  (i.e. beta>1/Tmin)
+
+  double betafreeze = betamax_ * sqrt(coolingFactor_);
+
+  while (beta < betafreeze) {
+    if(useTc_){
+      update(beta, tks,y, false, rho0);
+      while(merge(y, beta)){update(beta, tks, y, false, rho0);}
+      split(beta, tks, y);
+      beta=beta/coolingFactor_;
+    }else{
+      beta=beta/coolingFactor_;
+      splitAll(y);
+    }
+    
+    // make sure we are not too far from equilibrium before cooling further
+    niter = 0;
+    while ((update(beta, tks, y, false, rho0) > 1.e-6) && 
+	   (niter++ < maxIterations_)) {}
+
+    if(verbose_){ dump( beta, y, tks, 0); }
+  }
+  
+
+  if(useTc_){
+    //last round of splitting, make sure no critical clusters are left
+    if(verbose_){ std::cout << "last spliting at " << 1./beta << std::endl; }
+    update(beta, tks,y, false, rho0);// make sure Tc is up-to-date
+    while(merge(y,beta)){update(beta, tks,y, false, rho0);}
+    unsigned int ntry=0;
+    double threshold = 1.0;
+    while( split(beta, tks, y, threshold) && (ntry++<10) ){
+      niter=0; 
+      while((update(beta, tks,y, false, rho0)>1.e-6)  && (niter++ < maxIterations_)){}
+      while(merge(y,beta)){update(beta, tks,y, false, rho0);}
+      if(verbose_){ 
+	std::cout << "after final splitting,  try " <<  ntry << std::endl; 
+	dump(beta, y, tks, 2); 
+      }
+      // relax splitting a bit to reduce multiple split-merge cycles of the same cluster
+      threshold *= 1.1; 
+    }
+  }else{
+    // merge collapsed clusters 
+    while(merge(y,beta)){update(beta, tks,y, false, rho0);}  
+  }
+  
+  if (verbose_) {
+    update(beta, tks,y, false, rho0);
+    std::cout  << "dump after 1st merging " << endl;
+    dump(beta, y, tks, 2);
+  }
+  
+  
+  // switch on outlier rejection at T=Tmin
+  if(dzCutOff_ > 0){
+    rho0 = 1./nt;
+    for(unsigned int a=0; a<10; a++){ update(beta, tks, y, true, a*rho0/10);} // adiabatic turn-on
+  }
+
+  niter=0;
+  while ((update(beta, tks, y, true, rho0) > 1.e-8) && (niter++ < maxIterations_)) {};
+  if (verbose_) {
+    std::cout  << "dump after noise-suppression, rho0=" << rho0  << "  niter = " << niter << endl;
+    dump(beta, y, tks, 2);
+  }
+
+  // merge again  (some cluster split by outliers collapse here)
+  while (merge(y, beta)) {update(beta, tks, y, true, rho0); }
+  if (verbose_) {
+    std::cout  << "dump after merging " << endl;
+    dump(beta, y, tks, 2);
+  }
+
+  // go down to the purging temperature (if it is lower than tmin)
+  while( beta < betapurge_ ){
+    beta = min( beta/coolingFactor_, betapurge_);
+    niter = 0;
+    while ((update(beta, tks, y, false, rho0) > 1.e-8) && (niter++ < maxIterations_)) {}
+  }
+
+
+  // eliminate insigificant vertices, this is more restrictive at higher T
+  while (purge(y, tks, rho0, beta)) {
+    niter = 0;
+    while (( update(beta, tks, y, true, rho0) >  2.5e-7 * y.GetSize() ) && (niter++ < maxIterations_)) {
```

Comment:
is there any documentation for this magic number? (it looks like this was changed already since the original PR submission as a result of some tuning).